File: src/game/events/game/ScoreGameEvent.java
package game.events.game;

import game.Bomberman;
import game.events.models.GameEvent;

/**
 * Fires a game event using Strategy Pattern;
 */
public class ScoreGameEvent implements GameEvent {
    @Override
    public void invoke(Object arg) {
        Bomberman.getMatch().getCurrentLevel().onScoreGameEvent((int) arg);
    }
}

File: src/game/engine/events/game/ScoreGameEvent.kt
package game.engine.events.game

import game.Bomberman
import game.engine.events.models.GameEvent


class ScoreGameEvent : GameEvent {
    override fun invoke(arg: Any?) {
        Bomberman.getMatch().currentLevel!!.eventHandler.onScoreGameEvent(arg as Int)
    }
}